src/Compilers/CSharp/Test/Emit3/Semantics/PrimaryConstructorTests.cs

```diff
@@ -277,29 +311,66 @@ public void LanguageVersion_06(string typeKeyword, string baseKeyword)
 
 " + baseKeyword + @" Base{}
 ";
+            // For class inheriting from class without parameter list, we get the new specific error
+            // For other combinations (struct from interface), we get the generic error
+            var isClassInheritingFromClass = typeKeyword == "class" && baseKeyword == "class";
+            var errorCode = isClassInheritingFromClass
+                ? ErrorCode.ERR_UnexpectedArgumentListInBaseTypeWithoutParameterList
+                : ErrorCode.ERR_UnexpectedArgumentList;
+
             var comp = CreateCompilation(src1, parseOptions: TestOptions.Regular11, options: TestOptions.ReleaseDll);
-            comp.VerifyDiagnostics(
-                // (3,7): error CS8861: Unexpected argument list.
-                // : Base()
-                Diagnostic(ErrorCode.ERR_UnexpectedArgumentList, "()").WithLocation(3, 7),
-                // (4,1): error CS9058: Feature 'primary constructors' is not available in C# 11.0. Please use language version 12.0 or greater.
-                // ;
-                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion11, ";").WithArguments("primary constructors", "12.0").WithLocation(4, 1)
-                );
+            if (isClassInheritingFromClass)
+            {
+                comp.VerifyDiagnostics(
+                    // (3,7): error CS9339: Cannot pass arguments to the base type without a parameter list on the type declaration. Consider adding an empty parameter list to 'Point'.
+                    // : Base()
+                    Diagnostic(errorCode, "()").WithArguments("Point").WithLocation(3, 7),
+                    // (4,1): error CS9058: Feature 'primary constructors' is not available in C# 11.0. Please use language version 12.0 or greater.
+                    // ;
+                    Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion11, ";").WithArguments("primary constructors", "12.0").WithLocation(4, 1));
+            }
+            else
+            {
+                comp.VerifyDiagnostics(
+                    // (3,7): error CS8861: Unexpected argument list.
+                    // : Base()
+                    Diagnostic(errorCode, "()").WithLocation(3, 7),
```

Comment:
The comment two lines above is incorrect half of the time. Also applies in other verification below